### PR TITLE
Add user context

### PR DESCRIPTION
**Pull request by seeziyang**
- Add react context for user
- Fix wrong error handling in sign in and sign up
- Navigate to main page after sign in

### CODE REVIEW

File: src/pages/SignUpResearcher.js [hunk 1]
```diff
@@ -31,17 +32,21 @@ function SignUpResearcher() {
...
     return errors;
   };
 
-  const handleSubmit = (values) => {
+  const handleSubmit = values => {
     //TODO: authenticate to allow login
+
+    // Db.addResearchData(useContext(UserContext)?.uid, ...); TODO
+    // navigate to next page
+
```
Comment (seeziyang) [inline src/pages/SignUpResearcher.js:49]: @CedricLSM You can add pass in the additional data for researcher in here and similarly in the `handleSubmit` function of SignUpVolunteer

File: src/util/Database.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      .catch(error => console.error("Error adding user to DB: ", error));
+  }
+
+  static addResearchData(userId, data) {
+    // TODO
+  }
+
+  static addVolunteerData(userId, data) {
+    // TODO
+  }
```
Comment (seeziyang) [inline src/util/Database.js:21]: @CedricLSM and this part to put the data into the db

File: src/util/Database.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    firebase
+      .firestore()
+      .collection("users")
+      .doc(userId)
+      .set(userData)
+      .catch(error => console.error("Error adding user to DB: ", error));
+  }
+
+  static addResearchData(userId, data) {
+    // TODO
```
Comment (wennyyustalim) [inline src/util/Database.js:16]: nit: prefer TODOs with description

File: src/util/Database.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  static addVolunteerData(userId, data) {
+    // TODO
```
Comment (wennyyustalim) [inline src/util/Database.js:20]: ditto

File: src/pages/SignUpVolunteer.js [hunk 5]
```diff
@@ -31,17 +31,21 @@ function SignUpResearcher() {
...
 
     return errors;
   };
 
-  const handleSubmit = (values) => {
+  const handleSubmit = values => {
     //TODO: authenticate to allow login
+
+    // Db.addVolunteerData(useContext(UserContext)?.uid, ...); TODO
+    // navigate to next page
```
Comment (wennyyustalim) [inline src/pages/SignUpVolunteer.js:47]: nit: prefers to not leave commented codes in production code. just for future reference :)

**Pull request merged**